File: libraries/js/Tweenings/back.js
class Back {
    static SWING_BACK_DEFAULT = 1.70158;
    static SWING_MULTIPLICATOR = 1.525;

    static evaluate(tweeningBehaviour, t, b, c, d, arg1){
        switch(tweeningBehaviour) {
            case ETweeningBehaviour.IN:
                return Back.in_s(t,b,c,d,arg1);
            case ETweeningBehaviour.OUT:
                return Back.out_s(t,b,c,d,arg1);
            case ETweeningBehaviour.IN_OUT:
                return Back.inOut_s(t,b,c,d,arg1);
            default:
                return null;
        }
    }
    

    static in(t, b, c, d){
        var s = Back.SWING_BACK_DEFAULT;

		return c * (t /= d) * t *((s + 1) * t - s) + b;
    }


    static in_s(t, b, c, d, s){
        if (s == null)
            return Back.in(t, b, c, d);

        return c * (t /= d) * t *((s + 1) * t - s) + b;
    }


    static out(t, b, c, d){
        var s = Back.SWING_BACK_DEFAULT;

		return c * ((t = t / d - 1) * t * ((s + 1) * t + s) + 1) + b;
    }

    static out_s(t, b, c, d, s){
        if (s == null)
            return Back.out(t, b, c, d);

        return c * ((t = t / d - 1) * t *((s + 1) * t + s) + 1) + b;
    }

    
    static inOut(t, b, c, d){
        var s = Back.SWING_BACK_DEFAULT;

		if ((t /= d / 2) < 1) 
            return c / 2 * (t * t * (((s *= (Back.SWING_MULTIPLICATOR)) + 1) * t - s)) + b;

		return c / 2 * ((t -= 2) * t * (((s *= (Back.SWING_MULTIPLICATOR)) + 1) * t + s) + 2) + b;
    }

    static inOut_s(t, b, c, d, s){
        if (s == null)
            return Back.inOut(t, b, c, d);

        if ((t /= d / 2) < 1) 
            return c / 2 * (t * t * (((s *= (Back.SWING_MULTIPLICATOR)) + 1) * t - s)) + b;

		return c / 2 * ((t -= 2) * t * (((s *= (Back.SWING_MULTIPLICATOR)) + 1) * t + s) + 2) + b;
    }
}

File: libraries/python/Tweenings/back.py
from ETweeningBehaviour import ETweeningBehaviour as ETB

class Back:
    SWING_BACK_DEFAULT: float = 1.70158
    SWING_MULTIPLICATOR: float = 1.525

    def evaluate(tweeningBehaviour, t, b, c, d, arg1) :
        if tweeningBehaviour == ETB.IN:
            return Back.in_s(t,b,c,d,arg1)
        elif tweeningBehaviour == ETB.OUT:
            return Back.out_s(t,b,c,d,arg1)
        elif tweeningBehaviour == ETB.IN_OUT:
            return Back.inOut_s(t,b,c,d,arg1)

        return None
        

    def in_back(t, b, c, d) :
        s: float = Back.SWING_BACK_DEFAULT
        
        t = t / d

        return c * t * t *((s + 1) * t - s) + b


    def in_s(t, b, c, d, s):
        if (s == None) :
            return Back.in_back(t, b, c, d)

        t = t / d

        return c * t * t *((s + 1) * t - s) + b
    


    def out(t, b, c, d):
        s: float = Back.SWING_BACK_DEFAULT

        t = t = t / d - 1

        return c * (t * t * ((s + 1) * t + s) + 1) + b
    

    def out_s(t, b, c, d, s):
        if (s == None) :
            return Back.out(t, b, c, d)

        t = t / d - 1

        return c * (t * t *((s + 1) * t + s) + 1) + b
    

    
    def inOut(t, b, c, d):
        s: float = Back.SWING_BACK_DEFAULT * Back.SWING_MULTIPLICATOR

        t = t / (d / 2)

        if (t < 1):
            return c / 2 * (t * t * ((s + 1) * t - s)) + b

        t = t - 2

        return c / 2 * (t * t * ((s + 1) * t + s) + 2) + b
    

    def inOut_s(t, b, c, d, s):
        if (s == None) :
            return Back.inOut(t, b, c, d)

        t = t / (d / 2)
        s = s * Back.SWING_MULTIPLICATOR

        if (t < 1) :
            return c / 2 * (t * t * ((s + 1) * t - s)) + b

        t = t - 2

        return c / 2 * (t * t * ((s + 1) * t + s) + 2) + b
    